src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+import {action, computed, observable} from 'mobx';
+import React from 'react';
+import RectangleSelection from '../RectangleSelection';
+import RoundingNormalizer from '../RectangleSelection/dataNormalizers/RoundingNormalizer';
+import type {SelectionData} from '../RectangleSelection/types';
+import log from 'loglevel';
+import {observer} from 'mobx-react';
+import selectionStyles from './imageSectorSelection.scss';
+
+@observer
+export default class ImageSectorSelection extends React.PureComponent {
+    props: {
+        /** Determines the position at which the selection box is rendered at the beginning. */
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
+    @observable containerHeight: number;
+
+    naturalHorizontalToReal = (h: number) => h * this.imageResizedWidth / this.image.naturalWidth;
+    realHorizontalToNatural = (h: number) => h * this.image.naturalWidth / this.imageResizedWidth;
+    naturalVerticalToReal = (v: number) => v * this.imageResizedHeight / this.image.naturalHeight;
+    realVerticalToNatural = (v: number) => v * this.image.naturalHeight / this.imageResizedHeight;
+
+    naturalDataToReal(data: SelectionData): SelectionData {
+        return {
+            width: this.naturalHorizontalToReal(data.width),
+            height: this.naturalVerticalToReal(data.height),
+            left: this.naturalHorizontalToReal(data.left),
+            top: this.naturalVerticalToReal(data.top),
+        };
+    }
+
+    realDataToNatural(data: SelectionData): SelectionData {
+        return {
+            width: this.realHorizontalToNatural(data.width),
+            height: this.realVerticalToNatural(data.height),
+            left: this.realHorizontalToNatural(data.left),
+            top: this.realVerticalToNatural(data.top),
+        };
+    }
+
+    @action componentWillMount() {
```

Comment:
But is the `@action` necessary? Isn't the `action()` call two lines down enough?